exporter/kafkaexporter/pdata_marshaler.go

```diff
@@ -14,30 +14,68 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
 )
 
+// KeyableLogsMarshaler is an extension of the LgosMarshaler interface intended to provide partition key capabilities
+// for log messages
+type KeyableLogsMarshaler interface {
+	LogsMarshaler
+	Key()
+}
+
 type pdataLogsMarshaler struct {
 	marshaler plog.Marshaler
 	encoding  string
+	keyed     bool
+}
+
+// Key configures the pdataLogsMarshaler to set the message key on the kafka messages
+func (p *pdataLogsMarshaler) Key() {
+	p.keyed = true
```

Comment:
any chance we could set this earlier in constructor?

Reply:
I think we probably could but might require a larger refactor?  Perhaps it's worth doing separately and then could be done for metrics, traces and logs consistently?

Reply:
Or can we refactor first?

Reply:
I did some more reading through the code and I'm not sure if this is worth doing in this PR.  As far as I can tell from my reading there's a couple of reasons:
1. The marshaller and the exporter are created at different times, when the marshaler is created there doesn't seem to be a obvious path to the configuration
2. The exporter seems to be responsible for interpreting configuration, and it's when the exporter is created (and interprets configuration) that it makes the call to the already-existing marshaler to ask it to key it's output.

Is this a blocker in your opinion for this added functionality that helps bring it up to parity with metrics and traces?

Reply:
no problem, we can tidy after

Reply:
@atoulme, FYI this refactoring was done in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33230